integration/integration.cabal

```diff
@@ -13,7 +13,7 @@ build-type:    Custom
 custom-setup
   setup-depends:
     , base
-    , Cabal
+    , Cabal             >=3.12
```

Comment:
Why is this constraint needed?

Reply:
because otherwise cabal multi-repl doesn't work.

Reply:
should I put a note there?

Reply:
Isn't there only one version of Cabal available in the nix env? The project as is doesn't depend on cabal multi-repl right?

Reply:
in theory yes, in practice no, Cabal is a boot library and hence belongs to the GHC, hence nixpkgs installs two versions, 3.12 and 3.10 (which comes with GHC and cannot be removed), that then leads to proto-lens tool to select the wrong (old) version because it is actually constrained to Cabal <3.12 for no particular reason. It's all a bit messed up, if we had purely nixpkgs deps, then this wouldn't cause any issues because cabal doesn't actually care about how these deps have been installed...

Reply:
This is the reason why cabal multi-repl is not enabled by default in HLS. By fixing this for us though, it removes the bug wrt duplicate constraints and it also improves HLS because of multi-repl. I think adding this constraint (which is provisioned by the nix env anyway) is worth this improvement.